### ISSUE TITLE
If attribute value is set to `null` attribute should be removed

### ISSUE THREAD

**Issue by Gozala** [Matt-Esch/virtual-dom#323]
When `attributes.foo` is `null` virtual-dom sets attribute value to `"null"` while it should instead remove attribute.

**Comment by Zolmeister** [comment]
`null` is a value, try `undefined`
(though I do see your point, that null perhaps should also signal undefined)

**Comment by Gozala** [comment]
Yes `undefined` does work as expected, but never the less I would expect `null` to behave the same rather than change attribute to string `"null"`

**Comment by teohhanhui** [comment]
`undefined` only works when creating element, but not when diff-ing (it retains the previous value). And according to @Matt-Esch's original comment `null` should be used to remove attribute.

https://github.com/Matt-Esch/virtual-dom/issues/54#issuecomment-40285542
